package.json

```diff
@@ -103,6 +103,11 @@
     "underscore": "^1.8.3",
     "validator": "^9.1.1"
   },
+  "resolutions": {
+    "lodash": "4.17.21",
```

Comment:
is there a reason to use resolutions instead of updating the packages that use these?

Also, can you use `yarn why` pre-resolutions to show which package version we're coming from on each of these? it'd be nice to have an idea of if we're doing patch/minor/major resolution changes

Reply:
# Before

`cryptiles`

```
=> Found "cryptiles@3.1.2"
info Reasons this module exists
   - "jest#jest-cli#jest-config#jest-environment-jsdom#jsdom#request#hawk" depends on it
   - Hoisted from "jest#jest-cli#jest-config#jest-environment-jsdom#jsdom#request#hawk#cryptiles"
```

`handlebars`

```
=> Found "handlebars@4.5.3"
info Reasons this module exists
   - "jest#jest-cli#@jest#core#@jest#reporters#istanbul-reports" depends on it
   - Hoisted from "jest#jest-cli#@jest#core#@jest#reporters#istanbul-reports#handlebars"
```

`lodash`

```
=> Found "lodash@4.17.11"
info Has been hoisted to "lodash"
info Reasons this module exists
   - Hoisted from "@babel#generator#lodash"
   - Hoisted from "@babel#traverse#lodash"
   - Hoisted from "@babel#types#lodash"
   - Hoisted from "@babel#cli#lodash"
   - Hoisted from "@babel#core#lodash"
   - Hoisted from "@babel#node#lodash"
   - Hoisted from "react-styleguidist#lodash"
   - Hoisted from "@babel#preset-env#@babel#plugin-transform-block-scoping#lodash"
   - Hoisted from "@babel#node#@babel#register#lodash"
   - Hoisted from "@babel#core#@babel#traverse#lodash"
   - Hoisted from "@babel#generator#@babel#types#lodash"
   - Hoisted from "@babel#helper-split-export-declaration#@babel#types#lodash"
   - Hoisted from "@babel#traverse#@babel#types#lodash"
   - Hoisted from "@babel#core#@babel#types#lodash"
   - Hoisted from "@babel#preset-env#@babel#types#lodash"
   - Hoisted from "sinon#@sinonjs#samsam#lodash"
   - Hoisted from "eslint#table#lodash"
   - Hoisted from "@babel#preset-env#@babel#plugin-transform-classes#@babel#helper-define-map#lodash"
   - Hoisted from "@babel#preset-env#@babel#plugin-transform-modules-amd#@babel#helper-module-transforms#lodash"
   - Hoisted from "@babel#preset-env#@babel#plugin-transform-sticky-regex#@babel#helper-regex#lodash"
   - Hoisted from "@instacart#eslint-config#babel-eslint#@babel#traverse#lodash"
   - Hoisted from "@babel#core#@babel#helpers#@babel#types#lodash"
   - Hoisted from "react-styleguidist#webpack-dev-server#http-proxy-middleware#lodash"
   - Hoisted from "@instacart#eslint-config#babel-eslint#@babel#traverse#@babel#generator#lodash"
   - Hoisted from "babel-jest#babel-plugin-istanbul#istanbul-lib-instrument#@babel#generator#lodash"
   - Hoisted from "babel-jest#babel-plugin-istanbul#istanbul-lib-instrument#@babel#traverse#lodash"
   - Hoisted from "@babel#preset-react#@babel#plugin-transform-react-jsx#@babel#helper-builder-react-jsx#@babel#types#lodash"
   - Hoisted from "babel-jest#@types#babel__core#@types#babel__traverse#@babel#types#lodash"
   - Hoisted from "@babel#preset-env#@babel#plugin-transform-parameters#@babel#helper-call-delegate#@babel#types#lodash"
   - Hoisted from "@babel#preset-env#@babel#plugin-transform-classes#@babel#helper-define-map#@babel#types#lodash"
   - Hoisted from "@babel#preset-env#@babel#plugin-transform-modules-systemjs#@babel#helper-hoist-variables#@babel#types#lodash"
   - Hoisted from "@babel#preset-env#@babel#plugin-transform-modules-amd#@babel#helper-module-transforms#@babel#types#lodash"
   - Hoisted from "@babel#preset-env#@babel#plugin-transform-object-super#@babel#helper-replace-supers#@babel#types#lodash"
   - Hoisted from "babel-jest#babel-plugin-istanbul#istanbul-lib-instrument#@babel#traverse#@babel#generator#lodash"
   - Hoisted from "@babel#preset-env#@babel#plugin-proposal-async-generator-functions#@babel#helper-remap-async-to-generator#@babel#helper-wrap-function#@babel#types#lodash"
   - Hoisted from "babel-jest#babel-plugin-istanbul#istanbul-lib-instrument#@babel#template#@babel#types#lodash"
   - Hoisted from "babel-jest#babel-plugin-istanbul#istanbul-lib-instrument#@babel#traverse#@babel#types#lodash"
info Disk size without dependencies: "4.86MB"
info Disk size with unique dependencies: "4.86MB"
info Disk size with transitive dependencies: "4.86MB"
info Number of shared dependencies: 0
=> Found "webpack-bundle-analyzer#lodash@4.17.10"
info This module exists because "webpack-bundle-analyzer" depends on it.
info Disk size without dependencies: "4.86MB"
info Disk size with unique dependencies: "4.86MB"
info Disk size with transitive dependencies: "4.86MB"
info Number of shared dependencies: 0
=> Found "eslint#lodash@4.17.15"
info Reasons this module exists
   - "eslint" depends on it
   - Hoisted from "eslint#inquirer#lodash"
info Disk size without dependencies: "4.86MB"
info Disk size with unique dependencies: "4.86MB"
info Disk size with transitive dependencies: "4.86MB"
info Number of shared dependencies: 0
=> Found "enzyme-to-json#lodash@4.17.4"
info This module exists because "enzyme-to-json" depends on it.
info Disk size without dependencies: "4.86MB"
info Disk size with unique dependencies: "4.86MB"
info Disk size with transitive dependencies: "4.86MB"
info Number of shared dependencies: 0
=> Found "eslint-import-resolver-webpack#lodash@4.17.4"
info This module exists because "eslint-import-resolver-webpack" depends on it.
info Disk size without dependencies: "4.86MB"
info Disk size with unique dependencies: "4.86MB"
info Disk size with transitive dependencies: "4.86MB"
info Number of shared dependencies: 0
=> Found "builder#lodash@3.10.1"
info This module exists because "@instacart#radium#builder" depends on it.
info Disk size without dependencies: "2.12MB"
info Disk size with unique dependencies: "2.12MB"
info Disk size with transitive dependencies: "2.12MB"
info Number of shared dependencies: 0
=> Found "cheerio#lodash@4.17.5"
info This module exists because "enzyme#cheerio" depends on it.
info Disk size without dependencies: "4.86MB"
info Disk size with unique dependencies: "4.86MB"
info Disk size with transitive dependencies: "4.86MB"
info Number of shared dependencies: 0
=> Found "@instacart/radium#lodash@4.17.15"
info Reasons this module exists
   - "@instacart#radium#@babel#cli" depends on it
   - Hoisted from "@instacart#radium#@babel#cli#lodash"
   - Hoisted from "@instacart#radium#webpack#async#lodash"
info Disk size without dependencies: "4.86MB"
info Disk size with unique dependencies: "4.86MB"
info Disk size with transitive dependencies: "4.86MB"
info Number of shared dependencies: 0
=> Found "webpack-merge#lodash@4.17.10"
info This module exists because "react-styleguidist#webpack-merge" depends on it.
info Disk size without dependencies: "4.86MB"
info Disk size with unique dependencies: "4.86MB"
info Disk size with transitive dependencies: "4.86MB"
info Number of shared dependencies: 0
=> Found "inquirer#lodash@4.17.4"
info This module exists because "webpack-cli#inquirer" depends on it.
info Disk size without dependencies: "4.86MB"
info Disk size with unique dependencies: "4.86MB"
info Disk size with transitive dependencies: "4.86MB"
info Number of shared dependencies: 0
=> Found "react-docgen#lodash@4.17.4"
info Reasons this module exists
   - "react-styleguidist#react-docgen#async" depends on it
   - Hoisted from "react-styleguidist#react-docgen#async#lodash"
info Disk size without dependencies: "4.86MB"
info Disk size with unique dependencies: "4.86MB"
info Disk size with transitive dependencies: "4.86MB"
info Number of shared dependencies: 0
=> Found "react-dev-utils#lodash@4.17.10"
info Reasons this module exists
   - "react-styleguidist#react-dev-utils#inquirer" depends on it
   - Hoisted from "react-styleguidist#react-dev-utils#inquirer#lodash"
info Disk size without dependencies: "4.86MB"
info Disk size with unique dependencies: "4.86MB"
info Disk size with transitive dependencies: "4.86MB"
info Number of shared dependencies: 0
=> Found "request-promise-core#lodash@4.17.5"
info This module exists because "jest#jest-cli#jest-config#jest-environment-jsdom#jsdom#request-promise-native#request-promise-core" depends on it.
info Disk size without dependencies: "4.86MB"
info Disk size with unique dependencies: "4.86MB"
info Disk size with transitive dependencies: "4.86MB"
info Number of shared dependencies: 0
```

`immer`

```
=> Found "immer@1.7.2"
info Reasons this module exists
   - "react-styleguidist#react-dev-utils" depends on it
   - Hoisted from "react-styleguidist#react-dev-utils#immer"
```

`merge-deep`

```
=> Found "merge-deep@3.0.2"
info Reasons this module exists
   - "@svgr#cli#@svgr#plugin-prettier" depends on it
   - Hoisted from "@svgr#cli#@svgr#plugin-prettier#merge-deep"
   - Hoisted from "@svgr#cli#@svgr#plugin-svgo#merge-deep"
info Disk size without dependencies: "16KB"
info Disk size with unique dependencies: "72KB"
info Disk size with transitive dependencies: "308KB"
info Number of shared dependencies: 9
```

Reply:
can't upgrade all of those `lodash` packages and can't update `svgr` without errors at runtime.